yt_dlp/extractor/nytimes.py

```diff
@@ -59,204 +51,546 @@ def get_file_size(file_size):
+            'description': 'md5:9ff8b47acbaf7f3ca8c732f5c815be2e',
+            'timestamp': 1695960700,
+            'upload_date': '20230929',
+            'creator': 'Stephanie Nolen, Natalija Gormalova',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+            'duration': 1322,
         }
     }, {
-        'url': 'http://www.nytimes.com/2016/10/16/books/review/inside-the-new-york-times-book-review-the-rise-of-hitler.html',
+        'url': 'https://www.nytimes.com/2023/11/29/business/dealbook/kamala-harris-biden-voters.html',
+        'md5': '3eb5ddb1d6f86254fe4f233826778737',
         'info_dict': {
-            'id': '100000004709479',
-            'title': 'The Rise of Hitler',
-            'ext': 'mp3',
-            'description': 'md5:bce877fd9e3444990cb141875fab0028',
-            'creator': 'Pamela Paul',
-            'duration': 3475,
+            'id': '100000009202270',
+            'ext': 'mp4',
+            'title': 'Kamala Harris Defends Biden Policies, but Says ‘More Work’ Needed to Reach Voters',
+            'description': 'md5:de4212a7e19bb89e4fb14210ca915f1f',
+            'timestamp': 1701290997,
+            'upload_date': '20231129',
+            'uploader': 'By The New York Times',
+            'creator': 'Katie Rogers',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+            'duration': 97,
         },
         'params': {
+            # inconsistant md5
             'skip_download': True,
         },
     }, {
-        'url': 'http://www.nytimes.com/news/minute/2014/03/17/times-minute-whats-next-in-crimea/?_php=true&_type=blogs&_php=true&_type=blogs&_r=1',
+        # multiple videos in the same article
+        'url': 'https://www.nytimes.com/2023/12/02/business/air-traffic-controllers-safety.html',
+        'info_dict': {
+            'id': 'air-traffic-controllers-safety',
+            'title': 'Drunk and Asleep on the Job: Air Traffic Controllers Pushed to the Brink',
+            'description': 'A nationwide shortage of controllers has resulted in an exhausted and demoralized work force that is increasingly prone to making dangerous mistakes.',
+        },
+        'playlist_count': 3,
+
+    }, {
+        'url': 'https://www.nytimes.com/2023/12/02/business/media/netflix-squid-game-challenge.html',
         'only_matching': True,
     }]
 
-    def _extract_podcast_from_json(self, json, page_id, webpage):
-        podcast_audio = self._parse_json(
-            json, page_id, transform_source=js_to_json)
-
-        audio_data = podcast_audio['data']
-        track = audio_data['track']
+    def _real_extract(self, url):
+        page_id = self._match_id(url)
+        webpage = self._download_webpage(url, page_id).replace(':undefined', ':null')
+        article_json = traverse_obj(self._search_json(
+            r'window.__preloadedData\s=', webpage, 'media details', page_id), ('initialData', 'data', 'article'))
```

Comment:
- `article_json` can be fatal since we don't have anything to extract without it
- escape the dot and match any amount of whitespace before `=` in the pattern
- instead of replacing `:undefined` with `:null` when setting `webpage`, try `transform_source=js_to_json` in the `_search_json()` params, or else if necessary use `transform_source=lambda x: x.replace('undefined', 'null')`
```suggestion
        webpage = self._download_webpage(url, page_id)
        article_json = self._search_json(
            r'window\.__preloadedData\s*=', webpage, 'media details', page_id,
            transform_source=js_to_json)['initialData']['data']['article']
```